fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/DebuggerWithSymbolTables.java

```diff
@@ -233,10 +252,68 @@ public void onDone() {
                     "query", Objects.requireNonNull(queryAsString).substring(0, Math.min(queryAsString.length(), 30)),
                     "duration-in-ms", TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - state.getStartTs()),
                     "ticks", state.getCurrentTick()));
+
+            final var eventProtos = state.getEventProtos();
+            if (eventProtos != null) {
+                writeEventsDelimitedToFile(eventProtos);
+            }
+
+            final var prerecordedEventProtoIterator = state.getPrerecordedEventProtoIterator();
+            if (prerecordedEventProtoIterator != null) {
+                Verify.verify(!prerecordedEventProtoIterator.hasNext(),
+                        "There are more prerecorded events, there are only " + state.getEvents().size() + " actual events.");
+            }
         }
         reset();
     }
 
+    private static void writeEventsDelimitedToFile(final List<PEvent> eventProtos) {
+        try {
+            final var tempFile = File.createTempFile("events-", ".bin",
+                    new File(System.getProperty("java.io.tmpdir")));
```

Comment:
Is this not what the two-arg version of `File.createTempFile` does?

Reply:
changed that call.